cvxpylayers/torch/cvxpylayer.py

```diff
@@ -172,63 +169,60 @@ def forward(ctx, *params):
                         (i, str(ctx.device), str(p.device))
                     )
 
-            # check shapes of params
+            ctx.batch_sizes = []
+            for i, (p, q) in enumerate(zip(params, param_order)):
+                # check and extract the batch size for the parameter
+                # 0 means there is no batch dimension for this parameter
+                # and we assume the batch dimension is non-zero
+                if p.ndimension() == q.ndim:
+                    batch_size = 0
+                elif p.ndimension() == q.ndim + 1:
+                    batch_size = p.size(0)
+                    assert batch_size > 0
+                else:
+                    raise RuntimeError(
+                        "Invalid parameter size passed in. Expected "
+                        "parameter {} to have have {} or {} dimensions "
+                        "but got {} dimensions".format(
+                            i, q.ndim, q.ndim + 1, p.ndimension()))
+
+                ctx.batch_sizes.append(batch_size)
+
+                # validate the parameter shape
+                p_shape = p.shape if batch_size == 0 else p.shape[1:]
+                if not np.all(p_shape == param_order[i].shape):
+                    raise RuntimeError(
```

Comment:
ValueError here as well

Reply:
Updated here, and the other RuntimeErrors in this file